src/index.js

```diff
@@ -0,0 +1,325 @@
+const figlet = require("figlet");
+const gradient = require("gradient-string");
+const table = require("table");
+const Db = require("./db");
+const { dbOptions } = require("./utils/dbConfig.js");
+
+const config = {
+	// table package style settings
+	border: table.getBorderCharacters("honeywell"),
+};
+
+const {
+	mainQuestions,
+	employeeInfo,
+	departmentInfo,
+	deleteRecord,
+} = require("./utils/questions");
+
+const {
+	employeeQuery,
+	roleQuery,
+	departmentQuery,
+	employeeByManager,
+} = require("./utils/queries");
+
+const {
+	generateRoleChoices,
+	generateEmployeeChoices,
+	generateDepartmentChoices,
+} = require("./utils/utils.js");
+
+const init = async () => {
+	// Welcome message function that uses both figlet and gradient
+	function welcomeMessage() {
+		const welcomeMsg = `SQL  Employee  Management  System`;
+
+		figlet(welcomeMsg, (err, data) => {
+			console.log(gradient.pastel.multiline(data));
+		});
+	}
+
+	welcomeMessage();
+
+	try {
+		// Establishing a connection to the database
+		const db = new Db({
+			host: process.env.DB_HOST || "localhost",
+			user: process.env.DB_USER || "root",
+			password: process.env.DB_PASSWORD || "password",
+			database: process.env.DB_NAME || "company_db",
+		});
+
+		await db.start();
+
+		// While inProgress is true, ask the user questions. If user selects quit - progress changes to false
+
+		let inProgress = true;
+
+		while (inProgress) {
+			// first action question - what action would the user like to take?
+			// this question will be the main question loop - the users choice you will trigger one of the if statement blocks of code below
+			let { userInput } = await inquirer.prompt(mainQuestions);
+
+			// if user selects view all departments, then retrieve from database and display table
+			if (userInput === gradient.summer("View all departments")) {
+				const departments = await db.query(departmentQuery);
+				console.table(departments);
+			}
+
+			// if user selects view all roles, then retrieve from database and display table
+			if (userInput === gradient.summer("View all roles")) {
+				const roles = await db.query(roleQuery);
+				console.table(roles);
+			}
+
+			// if user selects view all employees, then retrieve from database and display table
+			if (userInput === gradient.summer("View all employees")) {
+				const employees = await db.query(employeeQuery);
+				console.table(employees);
+			}
+
+			// if user selects add a department, then give the user the choice to add a department name
+			if (userInput === gradient.summer("Add a department")) {
+				const { departmentName } = await inquirer.prompt(departmentInfo);
+
+				await db.query(
+					`INSERT INTO department (name) VALUES ('${departmentName}');`
+				);
+			}
+
+			// if user selects add a role, then give the user the choice to add a role name
+			if (userInput === gradient.summer("Add a role")) {
+				const departments = await db.query(departmentQuery);
+
+				const roleQuestions = [
```

Comment:
Could these questions be extracted out to a separate module to clean this file up a bit?